src/frontend/Sider/Behandling/VedtakOgBeregning/Barnetilsyn/InnvilgeVedtak/InnvilgeBarnetilsyn.tsx

```diff
@@ -39,32 +39,39 @@ const Knapp = styled(Button)`
 `;
 
 const initUtgifter = (
-    vedtak: InnvilgeVedtakForBarnetilsyn | undefined,
-    barnIBehandling: GrunnlagBarn[]
-) => (vedtak ? vedtak.utgifter : tomUtgiftPerBarn(barnIBehandling));
+    barnMedOppfylteVilkår: GrunnlagBarn[],
+    vedtak?: InnvilgeVedtakForBarnetilsyn
+): Record<string, Utgift[]> =>
+    barnMedOppfylteVilkår.reduce((acc, barn) => {
+        const utgiftForBarn = vedtak?.utgifter[barn.barnId];
```

Comment:
Denne virker ikke når det ikke finnes et vedtak fra før. Trenger optional chaining her
```suggestion
        const utgiftForBarn = vedtak?.utgifter?.[barn.barnId];
```